Board overlays migrated from Matter

interfer opened the pull request:
Changed place for boards overlays.
Now they will be in boards/riscv/

andriy-bilynskyy commented:
The following west manifest projects have been modified in this Pull Request:

| Name | Old Revision | New Revision | Diff |
| ---- | ------------ | ------------ |------|
| hal_telink | https://github.com/telink-semi/hal_telink/commit/439208c46aa98297066cba042c605770c07280d8 (develop) | https://github.com/telink-semi/hal_telink/pull/68 | [telink-semi/hal_telink#68/files](https://github.com/telink-semi/hal_telink/pull/68/files) |

*Note: This message is automatically posted and updated by the Manifest GitHub Action.*

interfer committed:
boards: riscv: tlsr9518adk80d: Add boards

- B91 USB dongle (retention/non-retention)
- B91 Mars (retention/non-retention)

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

andriy-bilynskyy commented:
It seems there are lot of stuffs to rework.
Already updated submodule hal/telink which seems fine.
But here lot of stuufs should be reworked:
1. The next samples:
- samples/boards/tlsr9518adk80d_retention/ 
- samples/boards/tlsr9518adk80d_mars_retention/
- samples/boards/tlsr9528a_retention/
Just code duplication. We need to simplify but doing everything more complicated.
Just create one sample samples/boards/tlsr9x_retention/ (it also touches .github build)
2. Inheritance all boards from  boards/riscv/tlsr9518adk80d/tlsr9518adk80d-common.dtsi
seems to be bad idea due to removing nodes (already observed troubles with aliases to non-existing nodes). Thats better to create such file for each board and use only retention inheretance.
3. In drivers mixing #if CONFIG_X and #if defined CONFIG_X let's do it in some common way,
using same order.

interfer committed:
boards: dts: refactored DTS, retention sample for B9x boards

Replaced three samples:
-tlsr9518adk80d_retention
-tlsr9518adk80d_mars_retention
-tlsr9528a_retention
with single "tlsr9x_retention" for all supported boards.

Refactored common DTS inheritance from "tlsr9518adk80d-common.dtsi"
by creating respective files for Mars, dongle
and left only retention inheritance for each.

Signed-off-by: Dmytro Huz <diman1436@gmail.com>

interfer committed:
drivers: restyled #if-definitions in Telink B9x drivers

Refactored drivers and SoC as there was some mixing "#if CONFIG_X" and
"#if defined CONFIG_X"

Signed-off-by: Dmytro Huz <diman1436@gmail.com>
